### PR TITLE
Remove gtest file from cpp library

### PR DESCRIPTION
**Pull request by zymap**
---

**Motivation**

PR #9072 introduces the gtest file and a new license header.
Actually, we can avoid introducing the new license header by
adding the macro in our code.

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

**Comment by merlimat** [comment]
A couple of notes on #9072: 

 1. That header file is just a tiny macro, we shouldn't need to import a file from gtest for that
 2. The macro is about tests and must not be exposed in public API

**Review by eolivelli — APPROVED** [review]
LGTM

One curiosity:
Aren't we running apache-rat at every PR validation?

**Comment by zymap** [comment]
We are not running the apache-rat so I added it at PR #9815

**Comment by zymap** [comment]
@merlimat @BewareMyPower Could you please take a look at this? Thanks

File: pulsar-client-cpp/lib/TestUtil.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+//
+// Created by yong on 2021/3/5.
+//
```
Comment (BewareMyPower) [inline pulsar-client-cpp/lib/TestUtil.h:3]: It should be replaced by the license header.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
Since the original `gtest_prod.h` has been removed, the pom.xml should remove following code as well.

```xml
 <exclude>pulsar-client-cpp/include/gtest/gtest_prod.h</exclude>
```

**Comment by BewareMyPower** [comment]
There're three places that included the `gtest_prod.h` before:

```
lib/UnAckedMessageTrackerEnabled.h:21:#include "gtest/gtest_prod.h"
lib/MultiTopicsConsumerImpl.h:21:#include "gtest/gtest_prod.h"
lib/PartitionedConsumerImpl.h:21:#include "gtest/gtest_prod.h"
```

You should replace it with `#include "TestUtils.h"`

**Comment by zymap** [comment]
@BewareMyPower I address your comments. Please take a look. thank you

**Review by BewareMyPower — APPROVED** [review]
LGTM

**Review by eolivelli — APPROVED** [review]
LGTM
can you please change the PR title and description.
We should cite that we are working on the CPP client tests.

**Review by merlimat — APPROVED** [review]

**Pull request merged**